als-server/js/src/main/scala/org/mulesoft/als/server/modules/diagnostic/JsCustomValidator.scala

```diff
@@ -4,15 +4,23 @@ import org.mulesoft.als.logger.Logger
 import org.mulesoft.als.server.modules.diagnostic.custom.AMFOpaValidator
 import org.mulesoft.als.server.wasm.{AmfWasmOpaValidator, Callback}
 
+import java.util.UUID
+import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
 class JsCustomValidator(override val logger: Logger) extends AMFOpaValidator {
 
   override def validateWithProfile(profile: String, data: String): Future[ValidationResult] = {
-    logger.debug("Starting custom validation", "JsCustomValidator", "validateWithProfile")
+    val uuid = UUID.randomUUID().toString
+    logger.debug(s"Starting custom validation $uuid", "JsCustomValidator", "validateWithProfile")
     val cb = new Callback[ValidationResult]
     AmfWasmOpaValidator.validate(profile, data, debug = false, cb.callback)
-    cb.future
+    cb.future.map(
+      r => {
+        logger.debug(s"Finished custom validation $uuid", "JsCustomValidator", "validateWithProfile")
```

Comment:
```
    cb.future.andThen {
      case Failure(exception) => logger.error("","","")
      case Success(value) =>
        logger.debug("","","")
    }
```
    
not possible?

Reply:
We could,  but the callback (cb) is already handling the fail case

https://github.com/aml-org/als/blob/2891a8cb0fdf0aa123bfa56a40a1bce3f6c47852/als-server/js/src/main/scala/org/mulesoft/als/server/wasm/WasmLoader.scala#L19-L27